### PR TITLE
Fixed issue with not working with newer node.js versions

### PR DESCRIPTION
**Pull request by leezhiwei**
Hi, I managed to fix the issues with node.js by removing `extends Module` in the various modules, and updated the dependencies. #78

### CODE REVIEW

**Comment by lostkagamine** [comment]
Thank you!

**Comment by leezhiwei** [comment]
I hope my work helps out the community ofc.

**Comment by leezhiwei** [comment]
Need some help testing significant node versions. I know 20+ works, 16 also works too, according to my screenshot below.

File: updatecar.pgsql [hunk 1]
```diff
@@ -0,0 +1,39 @@
+UPDATE "Car" SET
```
Comment (lostkagamine) [inline updatecar.pgsql]: What is this file even doing?

File: package.json [hunk 2]
```diff
@@ -15,21 +15,22 @@
...
   "devDependencies": {
     "@types/express": "^4.17.13",
+    "@types/long": "4.0.0",
     "@types/node": "^18.0.3",
-    "prisma": "^4.13.0",
+    "prisma": "^5.9.1",
     "protobufjs-cli": "^1.0.0",
     "ts-node": "^10.8.2",
-    "typescript": "^4.7.4"
+    "typescript": "^4.0"
```
Comment (lostkagamine) [inline package.json:23]: Why downgrade TypeScript???

File: package.json [hunk 3]
```diff
@@ -38,7 +39,7 @@
     "express": "^4.18.1",
     "form-urlencoded": "^6.0.6",
     "iconv-lite": "^0.6.3",
-    "long": "^5.2.0",
+    "long": "4.0.0",
```
Comment (lostkagamine) [inline package.json:42]: Don't downgrade packages if you don't know what you're doing.

File: src/modules/cars.ts [hunk 4]
```diff
@@ -12,7 +12,7 @@ import * as carFunctions from "./cars/functions";
 import * as terminal from "./terminal/check_car";
 
 
-export default class CarModule extends Module {
```
Comment (lostkagamine) [inline src/modules/cars.ts:15]: Why?

File: src/modules/util/common.ts [hunk 5]
```diff
@@ -87,8 +87,9 @@ export function sendResponse(message: Writer, res: Response, headers: string[])
         }
 
         // Send the response to the client
-        r.send(Buffer.from(end));
-    }
+        var resp = Buffer.from(end);
```
Comment (lostkagamine) [inline src/modules/util/common.ts:90]: This intermediate variable isn't needed. This change is unnecessary.

**Review by lostkagamine — CHANGES_REQUESTED** [review]
Some changes in this pull request are completely unnecessary.

File: updatecar.pgsql (hunk 1 above)
Comment (leezhiwei) [reply updatecar.pgsql]: Sorry. Was testing and removed it? Why is it still in the repo.

File: src/modules/cars.ts (hunk 4 above)
Comment (leezhiwei) [reply src/modules/cars.ts:15]: This one was the change that made it work, this line of code resolved it.

File: src/modules/util/common.ts (hunk 5 above)
Comment (leezhiwei) [reply src/modules/util/common.ts:90]: I agree it is. Sorry was debugging and accidentally left it in.

File: src/modules/cars.ts (hunk 4 above)
Comment (lostkagamine) [reply src/modules/cars.ts:15]: That's odd. Why wasn't it booting with the `extends` in place?

File: updatecar.pgsql (hunk 1 above)
Comment (lostkagamine) [reply updatecar.pgsql]: You can `git rm` it.

**Comment by leezhiwei** [comment]
Honestly speaking. I have no idea why the TS downgraded. All I did was a `npm i`

**Comment by leezhiwei** [comment]
Honestly speaking. I have no idea why the TS downgraded. All I did was a `npm i`

**Comment by leezhiwei** [comment]
@lostkagamine currently build (on this repo) of this results in this Long type error.

**Comment by leezhiwei** [comment]
Why my long was downgraded was due to this stackoverflow. https://stackoverflow.com/questions/58159115/error-ts2304-cannot-find-name-long-when-compiling-typescript-with-google-ap

**Comment by leezhiwei** [comment]
> For the future, please post text logs instead of screenshots, it makes it really hard to read.

Sorry. Used to screenshots as logs here in Asia... I'll do codeblock quotes.... Sorry.

**Comment by leezhiwei** [comment]
Next this error appears
```
lee@mainservlee8219 ~/Bayshore (master)> yarn dev
yarn run v1.22.19
$ tsc && node dist
Loading config file...
/home/lee/Bayshore/dist/modules/cars.js:13
        function __() { this.constructor = d; }
                                         ^

TypeError: Cannot set property constructor of [object Ob
ject] which has only a getter
    at new __ (/home/lee/Bayshore/dist/modules/cars.js:1
3:42)
    at __extends (/home/lee/Bayshore/dist/modules/cars.j
s:14:84)
    at /home/lee/Bayshore/dist/modules/cars.js:98:5
    at Object.<anonymous> (/home/lee/Bayshore/dist/modul
es/cars.js:302:2)
    at Module._compile (node:internal/modules/cjs/loader
:1198:14)
    at Object.Module._extensions..js (node:internal/modu
les/cjs/loader:1252:10)
    at Module.load (node:internal/modules/cjs/loader:107
6:32)
    at Function.Module._load (node:internal/modules/cjs/
loader:911:12)
    at Module.require (node:internal/modules/cjs/loader:
1100:19)
    at require (node:internal/modules/cjs/helpers:119:18
)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for docum
entation about this command.
```

**Comment by leezhiwei** [comment]
```
Loading config file...
TypeError: Cannot set property constructor of [object Ob
ject] which has only a getter
    at new __ (/home/lee/Bayshore/src/modules/cars.ts:13
:42)use('/', appRouter);
    at __extends (/home/lee/Bayshore/src/modules/cars.ts
:14:84)
    at /home/lee/Bayshore/src/modules/cars.ts:15:40
    at Object.<anonymous> (/home/lee/Bayshore/src/module
s/cars.ts:255:2)
    at Module._compile (node:internal/modules/cjs/loader
:1198:14)
    at Module.m._compile (/usr/lib/node_modules/ts-node/
src/index.ts:1618:23)ET / Mucha Server
    at Module._extensions..js (node:internal/modules/cjs
/loader:1252:10)                      120,25        71%
    at Object.require.extensions.<computed> [as .ts] (/u
sr/lib/node_modules/ts-node/src/index.ts:1621:12)
    at Module.load (node:internal/modules/cjs/loader:107
6:32)
    at Function.Module._load (node:internal/modules/cjs/
loader:911:12)
```
This is the error that is fixed by removing `extends Module`

**Comment by leezhiwei** [comment]
The `updatecar.pgsql` is not included in the latest commit. FYI.

**Comment by leezhiwei** [comment]
Hi guys?

**Comment by leezhiwei** [comment]
I'll try and rebase my code, without the sql script being in the history and will request another PR. @lostkagamine please close this PR while I reset my branch. This can be archived for informational sake.

**Comment by lostkagamine** [comment]
Thank you; closed as requested.

**Pull request closed without merging**